### PR TITLE
Fr edit fly

### PR DESCRIPTION
**Pull request by fredrondina**
### Issue(s) related to this PR:
* closes #
#10 #4 #3 #8 #9 

### Type of change made:
* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] Styling- no new features

#### If this submission is a new feature:
* [x] Have you added an explanation of what your changes do?
* [ ] Have you lint your code locally prior to submission?

### Change Log:
*Added endpoints for new, show, index, edit, and destroying of flies

### Were there any challenges that arose? If so, how were the addressed?
* NOPE

### Technical Debt:
* fish table still needs to be changed

### Testing:
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed?
* [ ] Have you run linter?

**Test coverage** = 100%

### CODE REVIEW

File: app/controllers/api/v1/flies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  def fly_params
+    params.require(:fly).permit(:name, :size, :color, :category, :amount, :favorited?)
+  end
+
+  def options
+    options = {}
+    options[:is_collection] = false
+    return options
+  end
+end
```
Comment (tylertomlinson) [inline app/controllers/api/v1/flies_controller.rb:36]: Will need to add 404 error to send in json as well. Can add to next pr

File: spec/models/fly_spec.rb [hunk 2]
```diff
@@ -5,10 +5,10 @@
...
-    it { should validate_presence_of(:type) }
+    it { should validate_presence_of(:category) }
   end
 
-  describe 'relationships' do
-    it { should belong_to(:user) }
-  end
+  # describe 'relationships' do
+  #   it { should belong_to(:user) }
+  # end
```
Comment (tylertomlinson) [inline spec/models/fly_spec.rb:13]: Tech debt here to discuss.. No users so the users flybox index will just be the fly index action.

File: spec/requests/api/v1/flies_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
```
Comment (tylertomlinson) [inline spec/requests/api/v1/flies_request_spec.rb:1]: I can add the sad path testing here as well

**Review by tylertomlinson — APPROVED** [review]

File: spec/models/fly_spec.rb (hunk 2 above)
Comment (fredrondina) [reply spec/models/fly_spec.rb:13]: Yes, we are focusing for core fly and fish functionality for now, will go back and add user functionality later

**Pull request merged**